contrib/pkg/installmanager/installmanager_test.go

```diff
@@ -454,3 +455,253 @@ last line with password in text`,
+			scriptTemplate: `#!/bin/bash
+exit 2`,
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		im := InstallManager{
+			LogLevel: "debug",
+		}
+		assert.NoError(t, im.Complete([]string{}))
+		result, err := runGatherScript(fakeBootstrapIP, test.scriptTemplate, &im)
+		if test.expectedError {
+			assert.Error(t, err, "expected error for test case %s", test.name)
+		} else {
+			data, err := ioutil.ReadFile(result)
+			assert.NoError(t, err, "error reading returned log file data")
+			assert.Equal(t, test.expectedLogData, string(data))
+
+			// cleanup saved/copied logfile
+			if err := os.RemoveAll(result); err != nil {
+				t.Logf("couldn't delete saved log file: %v", err)
+			}
+		}
+	}
+}
+func TestGetBootstrapIP(t *testing.T) {
+	tests := []struct {
+		name                      string
+		terraformStateFileContent string
+		expectedIP                string
+		expectedError             bool
+	}{
+		{
+			name:          "terraform file doesn't exist",
+			expectedError: true,
+		},
+		{
+			name:                      "get bootstrap IP from terraform file",
+			expectedIP:                "1.1.1.1",
+			terraformStateFileContent: testTerraformStateFile("1.1.1.1"),
+		},
+		{
+			name:                      "bad json terraform",
+			terraformStateFileContent: `{ "notModules": "notResources" }`,
+			expectedError:             true,
+		},
+		{
+			name:                      "bootstrap instance gone from terraform file",
+			terraformStateFileContent: testTerraformStateFileWithInstanceAndIP("notBootstrap", "1.1.1.1"),
+			expectedError:             true,
+		},
+	}
+
+	for _, test := range tests {
+		tempDir, err := ioutil.TempDir("", "installmanagersshtest")
+		assert.NoError(t, err, "Errored while setting up temp dir for test")
+		defer os.RemoveAll(tempDir)
+
+		im := InstallManager{
+			LogLevel: "debug",
+			WorkDir:  tempDir,
+		}
+		assert.NoError(t, im.Complete([]string{}))
+
+		if test.terraformStateFileContent != "" {
+			assert.NoError(t, ioutil.WriteFile(filepath.Join(tempDir, "terraform.tfstate"), []byte(test.terraformStateFileContent), 0644),
+				"errored creating test terraform state file")
+		}
+
+		result, err := getBootstrapIP(&im)
+
+		if test.expectedError {
+			assert.Error(t, err, "expected error for test case")
+		} else {
+			assert.NoError(t, err, "unexpected error for test case")
+
+			assert.Equal(t, test.expectedIP, result, "returned IP not what was expected")
+		}
+
+	}
+}
+
+func testTerraformStateFile(ip string) string {
+	return testTerraformStateFileWithInstanceAndIP("aws_instance.bootstrap", ip)
+}
+
+func testTerraformStateFileWithInstanceAndIP(instanceName, ip string) string {
+	return fmt.Sprintf(`{
+	"modules": [
+		{
+			"path": [
+				"root",
+				"bootstrap"
+			],
+			"resources": {
+				"%s": {
+					"primary": {
+						"attributes": {
+							"public_ip": "%s"
+						}
+					}
+				}
+			}
+		}
+	]
+}`, instanceName, ip)
+}
+
+func TestInstallManagerSSH(t *testing.T) {
```

Comment:
@joelddiaz I think we could simplify this test to limit it to testing your code (which basically parses ssh-agent output, sets env vars and calls ssh-add). You could supply alternate binaries for `ssh-agent` and `ssh-add` just to ensure that they are properly being called (and that you're properly parsing the output)

Reply:
we kind of already do that with the installer binary: https://github.com/openshift/hive/blob/34814cafa464c8040918a15767d758ce1a957c9a/contrib/pkg/installmanager/installmanager_test.go#L55-L64

Reply:
k. have some fake ssh-agent and ssh-add instead of relying on actual binaries